src/pubsub/ua_pubsub_networkmessage_json.c

```diff
@@ -494,7 +495,7 @@ NetworkMessage_decodeJsonInternal(UA_NetworkMessage *dst, CtxJson *ctx,
     dst->messageIdEnabled = entries[0].found;
     dst->publisherIdEnabled = entries[2].found;
     if(dst->publisherIdEnabled)
-        dst->publisherIdType = UA_PUBLISHERDATATYPE_STRING;
+        dst->publisherIdType = UA_PUBLISHERIDTYPE_STRING;
```

Comment:
No, it could still be a string...
Check the pubIdType to disambiguate string and uint64.

Reply:
extended with if clause for UINT64 publisherIdType